sdk/unity/Assets/Picovoice/Picovoice.cs

```diff
@@ -93,31 +88,52 @@ public class Picovoice : IDisposable
+        /// </param>
         /// </returns>
-        public static Picovoice Create(string keywordPath, Action wakeWordCallback,
-                                       string contextPath, Action<Inference> inferenceCallback,
-                                       string porcupineModelPath = null, float porcupineSensitivity = 0.5f,
-                                       string rhinoModelPath = null, float rhinoSensitivity = 0.5f)
+        public static Picovoice Create(
+            string accessKey,
+            string keywordPath,
+            Action wakeWordCallback,
+            string contextPath,
+            Action<Inference> inferenceCallback,
+            string porcupineModelPath = null,
+            float porcupineSensitivity = 0.5f,
+            string rhinoModelPath = null,
+            float rhinoSensitivity = 0.5f,
+            bool requireEndpoint = true)
         {
-            Porcupine porcupine = Porcupine.Create(keywordPaths: new List<string> { keywordPath },
-                                                   modelPath: porcupineModelPath,
-                                                   sensitivities: new List<float> { porcupineSensitivity });
+            try
+            {
+                Porcupine porcupine = Porcupine.FromKeywordPaths(accessKey,
+                    keywordPaths: new List<string> { keywordPath },
+                    modelPath: porcupineModelPath,
+                    sensitivities: new List<float> { porcupineSensitivity });
 
-            Rhino rhino = Rhino.Create(contextPath: contextPath,
-                                       modelPath: rhinoModelPath,
-                                       sensitivity: rhinoSensitivity);
+                Rhino rhino = Rhino.Create(accessKey,
+                    contextPath: contextPath,
+                    modelPath: rhinoModelPath,
+                    sensitivity: rhinoSensitivity,
+                    requireEndpoint: requireEndpoint);
+            
+                if (porcupine.FrameLength != rhino.FrameLength)
+                {
+                    throw new PicovoiceInvalidArgumentException("Porcupine and Rhino frame lengths are different");
+                }
 
-            if (porcupine.FrameLength != rhino.FrameLength)
-            {
-                throw new ArgumentException("Porcupine and Rhino frame lengths are different");
-            }
+                if (porcupine.SampleRate != rhino.SampleRate) 
+                {
+                    throw new PicovoiceInvalidArgumentException("Porcupine and Rhino sample rate are different");
+                }
 
-            if (porcupine.SampleRate != rhino.SampleRate) 
+                return new Picovoice(porcupine, wakeWordCallback, rhino, inferenceCallback);
+            }
+            catch (Exception ex)
             {
-                throw new ArgumentException("Porcupine and Rhino sample rate are different");
+                mapToPicovoiceException(ex);
```

Comment:
`mapToPicovoiceException` can return a PicovoiceException and then `throw mapToPicovoiceException(ex);` can be done and return null removed